crates/wasi-common/src/sys/unix/oshandle.rs

```diff
@@ -1,123 +1,59 @@
-use crate::entry::EntryRights;
-use crate::sys::oshandle::{AsFile, OsHandle, OsHandleExt};
-use crate::wasi::{types, RightsExt};
-use std::fs::{File, OpenOptions};
+use crate::sys::AsFile;
+use std::cell::Cell;
+use std::fs::File;
 use std::io;
 use std::mem::ManuallyDrop;
-use std::os::unix::prelude::{AsRawFd, FileTypeExt, FromRawFd, IntoRawFd, RawFd};
+use std::os::unix::prelude::{AsRawFd, FromRawFd, IntoRawFd, RawFd};
 
-pub(crate) use super::sys_impl::osfile::*;
+#[derive(Debug)]
+pub(crate) struct OsHandle(Cell<RawFd>);
```

Comment:
Should `OsHandle` have a name with `Raw` in it, since it wraps a `RawFd`/`RawHandle`? Maybe `RawOSHandle`?